### PR TITLE
mob_chat_db txt to yml

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: -

* **Description of Pull Request**: 

Simple suggestion of the conversion of mob_chat_db to yaml format.

### CODE REVIEW

File: db/mob_chat_db.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#   Dialog                  Text displayed by the monster.
+###########################################################################
+
+Header:
+  Type: MOB_CHAT_DB
+  Version: 1
+
+Body:
+  - Id: 1
+    Color: 0xFF0000
```
Comment (Lemongrass3110) [inline db/mob_chat_db.yml:36]: Sorry if I might be blind and one value is different, but maybe make it default 0xFF0000?

File: src/map/mob.cpp [hunk 2]
```diff
@@ -5522,59 +5518,60 @@ static bool mob_readdb_group(char* str[], int columns, int current){
...
 
-	if( ms == NULL ){
-		try{
-			ms = &mob_chat_db[msg_id];
-		}catch( const std::bad_alloc& ){
-			ShowError( "mob_parse_row_chatdb: Memory allocation for chat ID '%d' failed.\n", msg_id );
-			return false;
+	if (!exists) {
+		if (!this->nodesExist(node, { "Dialog", "Color" })) {
+			this->invalidWarning(node["Id"], "Missing node 'Dialog' or 'Color'.\n");
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:5541]: Not needed, will already be printed by nodesExist

File: src/map/mob.cpp [hunk 3]
```diff
@@ -5810,8 +5807,10 @@ static bool mob_parse_row_mobskilldb(char** str, int columns, int current)
 	else
 		ms->emotion = -1;
 
-	if(str[18] != NULL && mob_chat(atoi(str[18]))!=NULL)
-		ms->msg_id = atoi(str[18]);
+	if (*str[18]) {
+		uint16 id = static_cast<uint16>(atoi(str[18]));
+		ms->msg_id = ((mob_chat_db.find(id) != nullptr) ? id : 0);
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:5812]: If you already touch this:
replace atoi with strtol and give a proper error message if id is != 0 and not found please.

**Review by Lemongrass3110 — APPROVED** [review]

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/mob.cpp [hunk 4]
```diff
@@ -3854,7 +3854,7 @@ int mobskill_use(struct mob_data *md, t_tick tick, int event)
 		if ( ms[i]->msg_id ){ //Display color message [SnakeDrak]
 			std::shared_ptr<s_mob_chat> mc = mob_chat_db.find(ms[i]->msg_id);
 
-			if (mc) {
+			if (mc && mc->msg.length() > 0) {	// skip if empty
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:3857]: Wouldnt it be better to remove the entry from the database, if it is empty?
Also check out #5974.
This way users can remove an entry by "Remove: <message id>" if it is merged.

Comment (Atemo) [reply src/map/mob.cpp:3857]: I misread your PR sorry. I will revert my last change

**Review by aleos89 — APPROVED** [review]

**Pull request merged**